### ISSUE TITLE
Invalid syntax error when parsing thermostat goal temp when shut off

### ISSUE THREAD

**Issue by tisba** [jayme-github/fritzbox_smarthome_exporter#10]
Hey there!

I'm pretty new to this project, but I noticed a ton of error messages in my logs when running it:

```
2022/04/12 06:01:46 Unable to parse goal temperature of "Livingroom" : strconv.ParseFloat: parsing "OFF": invalid syntax
```

I'm not terribly familiar what best practises are with Prometheus metrics, but I'd like to be able to display on my Grafana dashboard that the thermostat is turned off. I can't use `fritzbox_thermostat_goal`, because it's not exported because of the parsing error. And I couldn't find any other exported metric that indicates that the thermostat is off. Any ideas?

Also, I was wondering, if this is really an error worth reporting, as this is an expected state (IMO).

**Comment by jayme-github** [comment]
Hi and thanks for reporting what is absolutely a bug.
I'm naively parsing trying to convert the values fritzctl returns there which obviously is nor very suitable: https://github.com/jayme-github/fritzctl/blob/2185c66d7f6fdadb43a25d4ed35e74abe7a35ff5/fritz/thermostat.go#L44-L46

Not sure about the right approach here. We could easily just return the raw values: 254 for "ON" (whatever that might mean exactly) and 253 for "OFF" but that might lead to confusion (when plotting the values for example). An alternative would be to add another metric (like `fritzbox_thermostat_state`) being 0/1 and just not collect/update the value of `fritzbox_thermostat_goal` in that case. What do you think?

Btw. the API specs (https://avm.de/fileadmin/user_upload/Global/Service/Schnittstellen/AHA-HTTP-Interface.pdf) say stat the same is expected for comfort and savings temperatures (as well as the measured temperature). But I don't see that happening.

**Comment by tisba** [comment]
I think having a "boolean" state (0/1) and not emitting a value for the goal should be the way to go.

This would be less confusing and more straight forward if you want to plot the data. If you care about the state, then you can also query that and display that like a "switch" – like you can already do with `fritzbox_switch_state`

![CleanShot 2022-04-13 at 19 36 54](https://user-images.githubusercontent.com/75156/163238281-7f2142cd-dcc5-4b3b-aa4f-52a6b419fcab.png)

**Comment by jayme-github** [comment]
This should be fixed with [v0.4.0](https://github.com/jayme-github/fritzbox_smarthome_exporter/releases/tag/v0.4.0). Please let me know if you run into any additional issues.
